### ISSUE TITLE
selectDates not working inside UIButton Action

### ISSUE THREAD

**Issue by REAOrlando** [patchthecode/JTAppleCalendar#590]
Hi Jay,

Thank you for this library( star added).  I've run into an issue while trying to implement a way to scroll to the next year.  I believe I'm almost there, but I can't get the `selectDates` function to set the date.  As it is this works for 1 iteration of the button action.  It will move ahead 1 year, but the selectedView does not show (become visible) on the calendarView like it does with the viewDidLoad() call of `calendarView.selectDates([Date()])`.  On repeated button taps it just starts over from the original date selected in the viewDidLoad.  If I could get it to actually select the new date then I believe my method should work with every tap of the button and each time I would get the same day selected just one year up.  Using the print functions below, they will both return the same date which seems like the date should be selected. So why doesn't it work.  

On a related note, have you considered adding a function to scroll up/down by year like you have for `scrollToSegment(.next)`? ps. I also tried a for loop x 12 with this and it was a mess.  

```swift
@IBAction func plusYearBtn(_ sender: UIButton) {
   let nextYear = userCalendar.date(byAdding: .year, value: 1, to: calendarView.selectedDates[0])!
   print(nextYear)
   calendarView.scrollToDate(next, animateScroll: false)
   calendarView.selectDates([next], triggerSelectionDelegate: true)
   print(calendarView.selectedDates[0])
```

I'm using the current master version in my Podfile found here :git => 'https://github.com/patchthecode/JTAppleCalendar.git', :branch => 'master'

**Comment by patchthecode** [comment]
checking this now.

**Comment by patchthecode** [comment]
I just checked my master branch with your code. And it works. Can you do a `pod update` on console and let me know after updating is done?

**Comment by REAOrlando** [comment]
As you probably expected, it was the nut behind the wheel!  I found out that I had mistakenly put an instance of `calendarView.selectDates([Date()])` in the `didScrollToDateSegmentWith` function.  So my code was working but then immediately the current date was reselected making it look like it wasn't.  I happened to use date formatter `formatter.date(from: "2017 10 18")!` instead of `Date()`.  This slowed it down just enough for me to see the selectedView flash briefly, so then I quickly found my error.  Thank you for confirming that my original code should work.
